### ISSUE TITLE
Set Hotmail to `OAuth2` instead of `Normal password`

### ISSUE THREAD

**Issue by rtanglao** [thunderbird/autoconfig#93]
* I created a new hotmail account and autoconfig set IMAP to `Normal password`
* This resulted in Hotmail not receiving email in TB unless I changed it to `OAuth2`
* I think the file to change is: https://autoconfig.thunderbird.net/v1.1/hotmail.com
* Until this is changed, we need to document it: https://github.com/thunderbird/knowledgebase-issues/issues/37

**Comment by rtanglao** [comment]
tl;dr not sure we can fix this. It seems like an MS issue not a Thunderbird issue. Willing as always to be proven wrong :-) !

* The howtogeek story:  [How To Geek: Thunderbird Has a Problem With Outlook, Hotmail, and Live Email Addresses - The problem stems from changes to Microsoft's authentication systems.](https://www.howtogeek.com/thunderbird-has-a-problem-with-outlook-hotmail-and-live-email-addresses/)is not 100% correct **at least in my experience.** You don't need an App password. i tested it as follows

1. In the web interface at outlook.com i created a new hotmail account, let's call it newhm@hotmail.com
2. i added newhm@hotmail.com
3. I set 2FA on with an email account (my gmail account as the second factor)
4. i added newhm@hotmail.com to TB
5. i couldn't send or receive email with newhm
6. i changed IMAP and SMTP to `OAuth2` (TB autoconfig set it to `Normal Password`
7. i tried to send and receive email with newhm and it worked (in the process firefox opened to the MS site and prompted me for the password, i entered my newhm password and also the 2FA code from gmail)

hooray for microsoft lack of clarity :-)
(i tested on Windows 11, Ubuntu and macOS all with the same results)

* Re: `Mozilla is working to resolve the problem. Presumably, Thunderbird needs a new OAuth setup with advanced MFA support. In the meantime, Thunderbird users must go down a circuitous route to receive emails from their Outlook, Hotmail, and Live inboxes. The steps are outlined in a [Mozilla support post](https://support.mozilla.org/en-US/kb/microsoft-oauth-authentication-and-thunderbird-202) and in several [Reddit threads](https://www.reddit.com/r/Thunderbird/comments/19fevs6/finally_found_the_fix_for_this_msft_email_issue/), but we'll give you the gist of it here.` <-- I respectfullly disagree. I don't think we need to fix anything in Thunderbird nor do users need to generate an App Password but I could be wrong.
* Re: `Unfortunately, this story is accompanied by an additional problem. Several email clients, including Thunderbird and the official Outlook app, cannot maintain a connection with Microsoft's email services. Logging in with an App Password works, but the account randomly disconnects. Microsoft is [investigating](https://support.microsoft.com/en-us/office/outlook-and-other-apps-are-unable-to-connect-to-outlook-com-f4202ebf-89c6-4a8a-bec3-3d60cf7deaef) the issue and explicitly mentions Thunderbird in its support post.` <-- I cannot reproduce this. Logging in with the outlook.com password, i.e. not the App Password works and I have had no issues with random disconnects in Thunderbird in my testing

**Comment by cketti** [comment]
Changing the config to use OAuth2 should be fine. I'll run some tests next week to make sure there's no problems.

**Comment by cketti** [comment]
@rtanglao: The updated config is now available in the staging environment. You can test it in Thunderbird by changing `mailnews.auto_config_url` to `https://autoconfig-stage.thunderbird.net/v1.1/`

It worked fine in my tests. But if you have the time, please do some checks on your side as well.

**Comment by rtanglao** [comment]
> https://autoconfig-stage.thunderbird.net/v1.1/

I tested this with thunderbird beta 123 and hotmail configured with OAuth2 for IMAP and SMTP so you can close this issue @cketti  Thanks for fixing it!

**Comment by cketti** [comment]
I now also pushed the change to the production environment.

**Comment by rtanglao** [comment]
* @cketti microsoft emailed us about this. See the email I forwarded you.
